core/src/main/java/hudson/Util.java

```diff
@@ -568,6 +568,25 @@ public static boolean isRelativePath(String path) {
     }
 
     /**
+     * A check if a file path is a descendant of a parent path
+     * @param forParent the parent the child should be a descendant of
+     * @param potentialChild the path to check
+     * @return true if so
+     * @throws IOException for invalid paths
+     * @since FIXME
+     * @see InvalidPathException
+     */
+    public static boolean isDescendant(File forParent, File potentialChild) throws IOException {
+        try {
+            Path child = potentialChild.getAbsoluteFile().toPath().normalize();
+            Path parent = forParent.getAbsoluteFile().toPath().normalize();
+            return child.startsWith(parent);
```

Comment:
This is not suitable for security purposes since it fails to canonicalize the paths and thus does not satisfy the goals of [JENKINS-26838](https://issues.jenkins-ci.org/browse/JENKINS-26838).